src/annotator/plugin/pdf.coffee

```diff
@@ -72,9 +72,9 @@ module.exports = class PDF extends Plugin
       # If the highlights are no longer in the document it means that either
       # the page was destroyed by PDF.js or the placeholder was removed above.
       # The annotations for these anchors need to be refreshed.
-      for hl in anchor.highlights
+      for hl, index in anchor.highlights
         if not document.body.contains(hl)
-          delete anchor.highlights
+          anchor.highlights.splice(index, 1)
```

Comment:
Note to self: I wondered how other code handled the fact that the anchor still has some highlights but no range. This is handled in [Guest#anchor](https://github.com/hypothesis/client/blob/2bc090cfa0fa471650b4e7b61622e79a8027ed1c/src/annotator/guest.coffee#L281) by adding all existing highlights to a list to remove when re-anchoring. The problem before was that the highlights that were still in the document were removed from `anchor.highlights` before `Guest#anchor` is called and therefore never got deleted.